src/components/WeatherApp.jsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useCallback, useEffect } from "react";
+import styles from "../styles/WeatherApp.module.css";
+import getForecast from "../helpers/getForecast";
+import LocationDetails from "./LocationDetails";
+import ForecastSummaries from "./ForecastSummaries";
+import ForecastDetails from "./ForecastDetails";
+import SearchForm from "./SearchForm";
+import ErrorMessage from "./ErrorMessage";
+
+export default function WeatherApp() {
+  const [city, setCity] = useState("");
+  const [location, setLocation] = useState({});
+  const [errorMessage, setErrorMessage] = useState("");
+  const [forecasts, setForecasts] = useState([]);
+  const [selectedDate, setSelectedDate] = useState(0);
+  const selectedForecast = forecasts.find(
+    (forecast) => forecast.date === selectedDate
+  );
+
+  const acknowledgeError = useCallback(() => {
+    setErrorMessage("");
+    setCity("");
+  }, []);
+
+  const handleForecastSelect = useCallback((date) => {
+    setSelectedDate(date);
+  }, []);
+
+  const updateCity = useCallback((event) => {
+    setCity(event.target.value);
+  }, []);
+
+  const submitCitySearch = useCallback(() => {
```

Comment:
All of these dont need to use `useCallback`, since you are just updating state values in each one. You can define them as plain arrow functions

Reply:
Hi Stu, without the useCallback hook, linting was complaining about passing functions as props. I was thinking of adding a rule to .eslint to ignore that...

Reply:
You shouldnt need to do that. I changed these to arrow functions locally and had no eslint errors at all